src/librustc/ty/maps.rs

```diff
@@ -929,6 +963,14 @@ define_maps! { <'tcx>
     [] needs_drop_raw: needs_drop_dep_node(ty::ParamEnvAnd<'tcx, Ty<'tcx>>) -> bool,
     [] layout_raw: layout_dep_node(ty::ParamEnvAnd<'tcx, Ty<'tcx>>)
                                   -> Result<&'tcx Layout, LayoutError<'tcx>>,
+
+    [] dylib_dependency_formats: MetaDataByCrateNum(CrateNum)
+                                    -> Rc<Vec<(CrateNum, LinkagePreference)>>,
+
+    [] is_allocator: MetaDataByCrateNum(CrateNum) -> bool,
+    [] is_panic_runtime: MetaDataByCrateNum(CrateNum) -> bool,
+
+    [] extern_crate: MetaDataByCrateNum(CrateNum) -> Rc<Option<ExternCrate>>,
```

Comment:
I would think that `CrateNum` would be preferred -- it seems clearer -- why do you think `DefId`, @eddyb?

One note: under the new dep-node system, a `DepNode(CrateNum)` is perfectly fine, and we should be able to handle it well (in the past, the "retracing" only worked for `DefId`, so we avoided using `CrateNum` in a `DepNode`.)